http-testkit-client/src/main/scala/com/wix/e2e/http/client/internals/RequestManager.scala

```diff
@@ -60,7 +61,28 @@ class NonBlockingRequestManager(request: HttpRequest) extends RequestManager[Fut
 class BlockingRequestManager(request: HttpRequest) extends RequestManager[HttpResponse] {
 
   def apply(path: String, but: RequestTransformer, withTimeout: FiniteDuration)(implicit baseUri: BaseUri): HttpResponse =
-    waitFor(nonBlockingRequestManager(path, but, withTimeout))(withTimeout + 1.second)
+    try
+      waitFor(nonBlockingRequestManager(path, but, withTimeout))(withTimeout + 1.second)
+    catch BlockingRequestManager.handleException
 
   private val nonBlockingRequestManager = new NonBlockingRequestManager(request)
 }
+
+object BlockingRequestManager {
```

Comment:
why not handle execption on https://github.com/wix/wix-http-testkit/blob/master/http-testkit-client/src/main/scala/com/wix/e2e/http/client/internals/RequestManager.scala#L32
seems like a better place for it

Reply:
There it is still the async flow, we don't have the caller stack there